### ISSUE TITLE
cmake-build issue with getaddrinfo

### ISSUE THREAD

**Issue by mback2k** [curl/curl#9313]
Example AppVeyor build: https://ci.appveyor.com/project/curlorg/curl/builds/44462657/job/relbmw0wb78k5t2s

```
C:/projects/curl/lib/curl_addrinfo.c: In function 'Curl_getaddrinfo_ex':
C:/projects/curl/lib/curl_addrinfo.c:192:5: error: implicit declaration of function 'freeaddrinfo'; did you mean 'getaddrinfo'? [-Werror=implicit-function-declaration]
  192 |     freeaddrinfo(aihead);
      |     ^~~~~~~~~~~~
      |     getaddrinfo
C:/projects/curl/lib/curl_addrinfo.c:192:5: error: nested extern declaration of 'freeaddrinfo' [-Werror=nested-externs]
C:/projects/curl/lib/curl_addrinfo.c: In function 'curl_dbg_getaddrinfo':
C:/projects/curl/lib/curl_addrinfo.c:556:14: error: 'getaddrinfo' undeclared (first use in this function); did you mean 'addrinfo'?
  556 |   int res = (getaddrinfo)(hostname, service, hints, result);
      |              ^~~~~~~~~~~
      |              addrinfo
C:/projects/curl/lib/curl_addrinfo.c:556:14: note: each undeclared identifier is reported only once for each function it appears in
cc1.exe: all warnings being treated as errors
make[2]: *** [lib/CMakeFiles/libcurl.dir/build.make:242: lib/CMakeFiles/libcurl.dir/curl_addrinfo.c.obj] Error 1
```

**Comment by bagder** [comment]
Duplicate of #9214
